Remove out of date known issues folder

gmarull opened the pull request:
It turns out that current known issues folder contains issues that are
no longer valid or present. I have searched for some regex
snippets listed, but I have not been able to find any. Documentation was
the last active user of the known-issues folder, but has moved to
another solution.

gmarull committed:
known-issues: delete out of date content

It turns out that current known issues folder contains issues that are
no longer valid or present. I have searched for some of the regex
snippets listed but I have not been able to find any. Documentation was
the last active user of the known-issues folder, but has moved to
another solution.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
ci: remove old paths in doc-build workflow

Documentation build still contained references to the old top Makefile
and old known-issues folder.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

nashif commented:
You need to remove reference to the folder in doc/README.rst and other places...

```
doc/CMakeLists.txt:set(KI_SCRIPT ${ZEPHYR_BASE}/scripts/filter-known-issues.py)
doc/CMakeLists.txt:set(CONF_DIR ${ZEPHYR_BASE}/.known-issues/doc)
doc/README.rst:``scripts/filter-known-issues.py`` together with a set of filter
doc/README.rst:configuration files in the ``.known-issues/doc`` folder.  (This
doc/README.rst:``.known-issues/doc`` folder, following the example of other conf files
doc/releases/release-notes-1.10.rst:* :github:`1542` - filter-known-issues.py fails if input file is empty
scripts/ci/twister_ignore.txt:.known-issues
scripts/filter-known-issues.py:                        default=[".known-issues/"],
scripts/filter-known-issues.py:path = ".known-issues/"
scripts/filter-known-issues.py:        print('''\n\n ---- New errors/warnings not tracked as .known-issues/, \
```

gmarull committed:
scripts: ci: twister_ignore: remove old reference to .known-issues

Remove old reference to the .known-issues folder.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented:
> You need to remove reference to the folder in doc/README.rst and other places...
> 
> ```
> doc/CMakeLists.txt:set(KI_SCRIPT ${ZEPHYR_BASE}/scripts/filter-known-issues.py)
> doc/CMakeLists.txt:set(CONF_DIR ${ZEPHYR_BASE}/.known-issues/doc)
> doc/README.rst:``scripts/filter-known-issues.py`` together with a set of filter
> doc/README.rst:configuration files in the ``.known-issues/doc`` folder.  (This
> doc/README.rst:``.known-issues/doc`` folder, following the example of other conf files
> doc/releases/release-notes-1.10.rst:* :github:`1542` - filter-known-issues.py fails if input file is empty
> scripts/ci/twister_ignore.txt:.known-issues
> scripts/filter-known-issues.py:                        default=[".known-issues/"],
> scripts/filter-known-issues.py:path = ".known-issues/"
> scripts/filter-known-issues.py:        print('''\n\n ---- New errors/warnings not tracked as .known-issues/, \
> ```

good catch, rebased and deleted remaining references, thanks!
